### DISCUSSION TITLE
Update expectation.

### DISCUSSION THREAD

**Discussion by DareDevil2000** [lencx/Noi#187] [General]
When will we expect to receive an update to this utility?

**Comment by kerzhaev** [comment]
Дорогой товарищ! Порадуй пожалуйста верных тебе последователей очередным обновлением.

**Comment by digisus** [comment]
Hello @lencx, I wanted to ask the same question. 

I am a frequent user of v0.4.0 on Debian. Observing how active this repo is with regular commits, I wonder why no new releases have been published since March 2024? Before releases came our regularly form what I can see...

Thank you for a brief info comment on this. And of course, a big thank you for providing such a useful tool for so many users!

**Reply by lencx** [reply]
Noi has paused updates since v0.4.0 because I am rewriting the project. This rewrite will optimize some details and introduce new features, such as:
- Switching from using a prompts JSON file to SQLite for storing prompts, allowing for more efficient data retrieval. Since prompts are an important personal asset for users, I want Noi to offer a more comprehensive solution for local prompt management.
- More powerful layout options (left and right sidebars, bottom toolbar, etc.). Demo: https://x.com/lencx_/status/1864577793243402541
- Noi PiP Mode (Picture-in-Picture), inspired by video playback’s picture-in-picture interaction. Users can run Noi in a small window when switching to the work screen, avoiding frequent window switches and minimizing workflow disruption. Demo: https://x.com/lencx_/status/1855883596986515939

**Reply by digisus** [reply]
Many thanks for your reply and great to hear you are working on a rewrite even! I wish you great success and looking forward to a new release whne you are ready. Do you have a rought time plan already when we can expect the new version? :pray:

**Reply by lencx** [reply]
The specific timeline is uncertain at the moment because some technical details may affect the speed of finding a solution. If all goes well, the new version should be released within the next two months.

**Comment by advmbarros** [comment]
GPT stopped working on my NOI.

It shows this error:

Oops!
We had a problem with your sign-in. Take a break and try again shortly.
